### ISSUE TITLE
AWSLoginRequest does not omit iam_request_headers when empty

### ISSUE THREAD

**Issue by adammw** [hashicorp/vault-client-go#147]
# Expected Behavior

AWSLoginRequest works for EC2 login.

# Current Behavior

AWSLoginRequest does not omit iam_request_headers when empty, leaving either a cryptic error of `Field validation failed: error converting input  for field "iam_request_headers": 'data' being decoded is nil`, or if set, prevents EC2 auth type due to `supplied auth values for both ec2 and iam auth types`.

# Failure Information

Vault 1.12.3
vault-client-go v0.2.0

## Steps to Reproduce

Call the AWS Login method for EC2:
```
	resp, err := client.Auth.AWSLogin(ctx, schema.AWSLoginRequest{
		Role:              role,
		Pkcs7:             strings.TrimSpace(strings.Replace(string(pkcs7), "\n", "", -1)),
		Nonce:             nonce,
	})
```

**Comment by averche** [comment]
Thanks for reporting this, @adammw! 

It looks like `iam_request_headers` is meant to be a map of headers but is marked as `string` in `openapi.json`. This might be an easy fix but I'll need to test to make sure.

**Comment by averche** [comment]
I've done a bit more digging into this issue. It seems that the root cause is that the OpenAPI-generated request fields are values rather than pointers and, when encoded into JSON, are sent as empty strings rather than omitted altogether. There are a few solutions that we are considering but they are probably not trivial and may take some time to implement correctly for all generated POST requests. In the meantime, there are a couple workarounds:

1. Use the generic `client.Write` instead:
   ```go
   resp, err := client.Write(ctx, "/auth/aws/login", map[string]any{
	   "data": map[string]any{
		   "role":   role,
		   "pkcs7":  strings.TrimSpace(strings.Replace(string(pkcs7), "\n", "", -1)),
		   "nonce":  nonce,
	   },
   })
   ```
2. For production use-cases, please use the established [vault/api](https://github.com/hashicorp/vault/tree/main/api) with [aws auth](https://github.com/hashicorp/vault/tree/main/api/auth/aws) instead.

**Comment by adammw** [comment]
would adding `omitempty` to the json field not suffice?

**Comment by averche** [comment]
Yes, that's the solution we are leaning towards. Unfortunately the generated structs have a custom generated `MarshalJSON()` which overrides that behavior (e.g for [`AWSLoginRequest`](https://github.com/hashicorp/vault-client-go/blob/v0.3.0/schema/model_aws_login_request.go#L51-L65)). We'll need to test whether it's safe to switch to default encoding. There is also a possibility where empty strings might be desired in some requests, though I can't think of any examples at the moment.

**Comment by averche** [comment]
Hi @adammw,

The #163 commit should have resolved this issue, could you please give it another try?

**Comment by averche** [comment]
Hi @adammw,

I have released a fix for this in [v0.3.1](https://github.com/hashicorp/vault-client-go/releases/tag/v0.3.1), please try it out and let us know if it works.
